src/com/owncloud/android/lib/resources/files/ChunkedUploadRemoteFileOperation.java

```diff
@@ -79,8 +87,26 @@ protected int uploadFile(OwnCloudClient client) throws HttpException, IOExceptio
                 ((ChunkFromFileChannelRequestEntity) mEntity).setOffset(offset);
                 mPutMethod.setRequestEntity(mEntity);
                 status = client.executeMethod(mPutMethod);
+
+                // TODO: Detect INVALID_CHARACTER_DETECT_IN SERVER in a better way?
```

Comment:
Please, remove TODO comment; and remove option for error 500, that was already fixed.